### ISSUE TITLE
Analyzer error with false Objects should not be disposed

### ISSUE THREAD

**Issue by OnurGumus** [SonarSource/sonar-csharp#649]
### Description

SonarLint gives false error of Objects should not be disposed while the Analyzer gives exception

### Repro steps

1. Create a new Console application

2. Post below code:

```
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApp16
{
    class Program
    {
        static void Main(string[] args)
        {
            IDisposable bmp = null;
            try
            {
                bmp = new SqlConnection();
                Console.Write(bmp.GetHashCode());
            }
            finally
            {
                bmp?.Dispose();
            }
        }
    }
}
```

### Expected behavior

This is a legit code. There shouldn't be any warnings at least regarding disposing.

### Actual behavior

Getting below warning:
```
Warning	AD0001	Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw an exception of type 'System.InvalidCastException' with message 'Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberBindingExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax'.'.	ConsoleApp16		1	Active	Analyzer 'SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce' threw the following exception:
'Exception occurred with following context:
Compilation: ConsoleApp16
SyntaxTree: c:\users\onur.gumus\documents\visual studio 2017\Projects\ConsoleApp16\ConsoleApp16\Program.cs
SyntaxNode: static void Main(string[] args) ... [MethodDeclarationSyntax]@[214..530) (11,8)-(23,9)

System.InvalidCastException: Unable to cast object of type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberBindingExpressionSyntax' to type 'Microsoft.CodeAnalysis.CSharp.Syntax.MemberAccessExpressionSyntax'.
   at SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce.ObjectDisposedPointerCheck.VisitInvocationExpression(InvocationExpressionSyntax instruction, ProgramState programState)
   at SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce.ObjectDisposedPointerCheck.PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
   at SonarAnalyzer.Helpers.FlowAnalysis.CSharp.ExplodedGraph.VisitInstruction(ExplodedGraphNode node)
   at SonarAnalyzer.Helpers.FlowAnalysis.Common.ExplodedGraph.Walk()
   at SonarAnalyzer.Rules.CSharp.ObjectsShouldNotBeDisposedMoreThanOnce.CheckForMultipleDispose(ExplodedGraph explodedGraph, SyntaxNodeAnalysisContext context)
   at SonarAnalyzer.Helpers.FlowAnalysis.CSharp.FlowAnalysisExtensions.Analyze(CSharpSyntaxNode declarationBody, ISymbol symbol, Action`2 analyze, SyntaxNodeAnalysisContext context)
   at SonarAnalyzer.Helpers.FlowAnalysis.CSharp.FlowAnalysisExtensions.<>c__DisplayClass0_0.<RegisterExplodedGraphBasedAnalysis>b__1(SyntaxNodeAnalysisContext c)
   at SonarAnalyzer.Helpers.DiagnosticAnalyzerContextHelper.<>c__DisplayClass0_0`1.<RegisterSyntaxNodeActionInNonGenerated>b__0(SyntaxNodeAnalysisContext c)
   at SonarAnalyzer.Helpers.SonarAnalysisContext.<>c__DisplayClass12_0`1.<RegisterSyntaxNodeAction>b__0(SyntaxNodeAnalysisContext c)
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__DisplayClass42_1`1.<ExecuteSyntaxNodeAction>b__1()
   at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock(DiagnosticAnalyzer analyzer, Action analyze, Nullable`1 info)
-----
'.

```

### Known workarounds

No
### Related information

* SonarC# Version: Using Sonar Lint latest from market place.
* Visual Studio Version 2017 Enterprise RTM with all updates latest.

**Comment by Evangelink** [comment]
Hi @OnurGumus,
Thanks for reporting this issue. It should be fixed by #587.

**Comment by OnurGumus** [comment]
How can I get  or build newer version of the visual studio plugin ?
